cfu-service/src/splitter.rs

```diff
@@ -0,0 +1,205 @@
+    fn resolve_content_response(&self, content_responses: &[FwUpdateContentResponse]) -> FwUpdateContentResponse;
+}
+
+/// Splitter struct
+pub struct Splitter<'a, C: Customization> {
+    /// CFU device
+    cfu_device: CfuDevice,
+    /// Component ID for each individual device
+    devices: &'a [ComponentId],
+    /// Customization for the Splitter
+    customization: C,
+}
+
+/// Maximum number of devices supported
+pub const MAX_SUPPORTED_DEVICES: usize = 8;
+
+impl<'a, C: Customization> Splitter<'a, C> {
+    /// Create a new Splitter
+    pub fn new(component_id: ComponentId, devices: &'a [ComponentId], customization: C) -> Self {
+        Self {
+            cfu_device: CfuDevice::new(component_id),
+            devices,
+            customization,
+        }
+    }
+
+    /// Create a new invalid FW version response
+    fn create_invalid_fw_version_response(&self) -> InternalResponseData {
+        let dev_inf = FwVerComponentInfo::new(FwVersion::new(0xffffffff), self.cfu_device.component_id());
+        let comp_info: [FwVerComponentInfo; MAX_CMPT_COUNT] = [dev_inf; MAX_CMPT_COUNT];
+        InternalResponseData::FwVersionResponse(GetFwVersionResponse {
+            header: GetFwVersionResponseHeader::new(1, GetFwVerRespHeaderByte3::NoSpecialFlags),
+            component_info: comp_info,
+        })
+    }
+
+    /// Create an offer rejection response
+    fn create_offer_rejection() -> InternalResponseData {
+        InternalResponseData::OfferResponse(FwUpdateOfferResponse::new_with_failure(
+            HostToken::Driver,
+            OfferRejectReason::InvalidComponent,
+            OfferStatus::Reject,
+        ))
+    }
+
+    /// Create a content rejection response
+    fn create_content_rejection(sequence: u16) -> InternalResponseData {
+        InternalResponseData::ContentResponse(FwUpdateContentResponse::new(
+            sequence,
+            CfuUpdateContentResponseStatus::ErrorInvalid,
+        ))
+    }
+
+    /// Process a fw version request
+    async fn process_get_fw_version(&self) -> InternalResponseData {
+        let mut versions = [GetFwVersionResponse {
+            header: Default::default(),
+            component_info: Default::default(),
+        }; MAX_SUPPORTED_DEVICES];
+
+        if self.devices.len() > MAX_SUPPORTED_DEVICES {
+            error!("More devices than supported");
+            return self.create_invalid_fw_version_response();
+        }
+
+        for (device_id, version) in zip(self.devices.iter(), versions.iter_mut()) {
+            if let Ok(InternalResponseData::FwVersionResponse(version_info)) =
+                cfu::route_request(*device_id, RequestData::FwVersionRequest).await
+            {
+                *version = version_info;
+            } else {
+                error!("Failed to get FW version for device {}", device_id);
+                return self.create_invalid_fw_version_response();
+            }
+        }
```

Comment:
This `zip()` is mostly repeated in several places. How about adding a helper method that takes a closure for the loop body and two iterators for the `zip` call? Something like (won't compile)a;

```rust
fn helper(a: A<T>, b: B<U>, body: FnOnce....) -> Result<...> {
    for (foo, bar) in zip(a, b) {
        body(...)
    }

    Ok(...)
}